### DISCUSSION TITLE
Recommended Display / Screen settings?? - Odd output occurring

### DISCUSSION THREAD

**Discussion by steffanvigano** [juftin/camply#225] [Q&A]
Just getting going, but I'm excited to have found this.  Thanks for developing!   I'm wondering about the formatting of the output.  Help screen outputs all sorts of odd characters that I assume is supposed to be painting a box around the text.   When I run a couple of the sample commands, It outputs some of it fine but then suddenly cursor jumps to the top of the screen and start painting over existing text.  Are there any recommended settings that I'm not finding?   I'm running this on Ubuntu Server 22.04 LTS and using SecureCRT as my ssh client.  Cheers.

**Comment by juftin** [comment]
Oh bummer. `camply` uses [rich](https://github.com/Textualize/rich) to provide this formatting. It works fine on most terminals - you might be the exception here. Sorry about that. You can set the `CAMPLY_LOG_HANDLER` environment variable to `PYTHON` (not case-sensitive) to get this to play nicely in your terminal.

**Reply by juftin** [reply]
I'll make sure to document this soon - thanks for raising this!

**Comment by steffanvigano** [comment]
Thanks for the prompt reply.   Sadly this didn't do anything to fix the issue.   Unless I did it wrong:

      export CAMPLY_LOG_HANDLER=PYTHON

**Reply by juftin** [reply]
Gotcha, so setting `CAMPLY_LOG_HANDLER=PYTHON` won't actually change those borders. I can look into updating that. It will affect the use of colors in the help messages and the formatting of logging too: 
<img width="786" alt="image" src="https://user-images.githubusercontent.com/49741340/227598519-0bdcb6a8-6dbd-41a1-a79b-942b62b21eb7.png">

rich does dome weird stuff with its logging where it figures out how big your screen is to handle word wrapping but it shouldn't be doing any overwriting of existing terminal output. When you set that environment variable the rich log handler isn't used at all - so I don't imagine that's causing you problems.

Does all this behavior still remain if you use docker on top of camply?

```shell
docker run --rm juftin/camply camply campgrounds --search "Glacier National Park"
```

**Reply by juftin** [reply]
camply uses emojis in its logging, I always imagined that the worst case scenario was that someone sees a `\u26FA` instead of ⛺️. 

I'm wondering if the emojis are causing carriage return or something like that. Is it possible to [install a nerd font](https://gist.github.com/matthewjberger/7dd7e079f282f8138a9dc3b045ebefa0) to see if that resolves your issue?

**Reply by steffanvigano** [reply]
I added a nerd font (just picked a random one since there were so many)
- No Change

Changed default shell from Bash to CSH
- No Change

Installed and ran through Docker
- No change. 

Sample output from docker (which looks the same as without):

![image](https://user-images.githubusercontent.com/4333672/227621783-822debc8-4f96-4286-b648-73ac5b33c3c9.png)

**Reply by steffanvigano** [reply]
Update... I fired up a copy of Putty and connected to that host and all is well.  It's got to be something with SecureCRT and it's emulation.  I tried all the various emulation modes available and could not get it to work cleanly.  I love SecureCRT, but will happily use Putty to config this wonderful tool.  Up to you if you want to keep troubleshooting it or not.  I'm happy to help you further if you have other ideas and have gone down the rabbit hole.  Cheers.
